src/auth/oauth2client.ts

```diff
@@ -491,6 +507,7 @@ export class OAuth2Client extends AuthClient {
   protected refreshTokenPromises = new Map<string, Promise<GetTokenResponse>>();
   readonly endpoints: Readonly<OAuth2ClientEndpoints>;
   readonly issuers: string[];
+  private client_authentication: ClientAuthentication;
```

Comment:
Looks like this should be readonly if we follow the existing convention

Reply:
Done